### DISCUSSION TITLE
How to: Only allow whitelisted Email-Adresses to fill out a form (without exposing the Email-Adresses on the frontend)?

### DISCUSSION THREAD

**Discussion by NilsDannemann** [supabase/supabase#7100] [Questions]
Hey everyone!

I have a simple "signup with magic link" form (in an nuxt3 app). 
But only people who are part of our newsletter email-list should be allowed to sign up.

How do I only allow whitelisted Email-Adresses be allowed to fill out this form without exposing the full email list on the frontend?

I'm not sure how to approach this.

Kind Regards,
Nils

**Comment by GaryAustin1** [comment]
If you have a table with the email addresses then you could lock it with RLS from all access.  Then have a security definer function called thru rpc, with an email address as the parameter, do a lookup and return true or false or other info if the email exists.  This method means someone could probe your email list brute force, one call at a time with emails, but probably not a concern to worry about for most apps.  

You could also enforce the auth signup by checking the user email in a before insert trigger function on auth.users.  If the email is not in your table then you just return null on the trigger to fail the user being added.

**Comment by NilsDannemann** [comment]
Thank you @GaryAustin1, I'll check that out.

For everyone needing an answer to this question:
I've also asked that question on a live stream with the dev-team:
https://youtu.be/bBV9kpfe9AM?t=918

Hope it helps

**Reply by bobby-alain** [reply]
Hi @NilsDannemann I am looking for the same answer, did you get to make it?

**Reply by NilsDannemann** [reply]
No, we ended up using another solution. But the approached in the video sounded promising

**Reply by bobby-alain** [reply]
another solution inside the supabase or with anther service?

**Reply by NilsDannemann** [reply]
Completely different service + we ended up changing the concept a little. So our solution doesnt really apply to this question anymore :D
